### PR TITLE
[4.0] Change com_fields to services

### PR DESCRIPTION
**Pull request by laoneo**
Moves com_fields to the service based extension setup.

See #20217 for details.

### CODE REVIEW

File: components/com_fields/Dispatcher/Dispatcher.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * @package     Joomla.Site
+ * @subpackage  com_fields
+ *
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\Fields\Site\Dispatcher;
```
Comment (wilsonge) [inline components/com_fields/Dispatcher/Dispatcher.php:10]: Given this is documented as basically the public API to the component, should we move this to the libraries folder? https://docs.joomla.org/J3.x:Adding_custom_fields/Implement_into_your_component#Public_helper_classes_and_API

Genuinely not sure if it's a good or bad idea - but I guess if we're namespacing it nows the time to do it if we're going to....

Comment (laoneo) [reply components/com_fields/Dispatcher/Dispatcher.php:10]: It uses models from com_fields so I would not move stuff to libraries with a dependency to an extension (I know we did that in the past).

**Comment by wilsonge** [comment]
Sorry can you fix the conflicts here and I'll merge this

**Comment by laoneo** [comment]
Done

**Pull request merged**